includes/class-amp-post-type-support.php

```diff
@@ -88,8 +88,9 @@ public static function get_support_errors( $post ) {
 			$errors[] = 'post-type-support';
 		}
 
-		if ( post_password_required( $post ) ) {
-			$errors[] = 'password-protected';
+		// Show password form instead of the post content.
+		if ( ! current_theme_supports( 'amp' ) && post_password_required( $post ) ) {
+			add_filter( 'the_content', [ __CLASS__, 'show_password_form' ] );
```

Comment:
I don't think this isn't really the right place for this filter to be added. Adding a filter is introducing global side effects which don't seem necessarily expected.

Reply:
I think this can be simplified quite a bit: 8519e4b

Reply:
Ah, that is indeed a much better solution. Thanks :ok_hand: